sros2/sros2/api/__init__.py

```diff
@@ -189,12 +214,21 @@ def create_ecdsa_param_file(path):
     run_shell_command('%s ecparam -name prime256v1 > %s' % (openssl_executable, path))
 
 
-def create_ca_key_cert(ecdsa_param_path, ca_conf_path, ca_key_path, ca_cert_path):
-    openssl_executable = find_openssl_executable()
-    check_openssl_version(openssl_executable)
-    run_shell_command(
-        '%s req -nodes -x509 -days 3650 -newkey ec:%s -keyout %s -out %s -config %s' %
-        (openssl_executable, ecdsa_param_path, ca_key_path, ca_cert_path, ca_conf_path))
+def create_ca_key_cert(ca_key_out_path, ca_cert_out_path):
+    private_key = ec.generate_private_key(ec.SECP256R1, cryptography_backend())
+    write_key(private_key, ca_key_out_path)
+
+    common_name = x509.NameAttribute(x509.oid.NameOID.COMMON_NAME, u'sros2testCA')
+    builder = x509.CertificateBuilder()\
+        .issuer_name(x509.Name([common_name])) \
+        .serial_number(x509.random_serial_number()) \
+        .not_valid_before(datetime.datetime.today() - datetime.timedelta(days=1)) \
+        .not_valid_after(datetime.datetime.today() + datetime.timedelta(days=3650)) \
+        .public_key(private_key.public_key()) \
+        .subject_name(x509.Name([common_name])) \
+        .add_extension(x509.BasicConstraints(ca=True, path_length=1), critical=True)
```

Comment:
Cryptography's tutorials establish a pattern that I prefer over the line continuations:

```python3
x509.CertificateBuilder().subject_name(
    subject
).issuer_name(
    issuer
)
# ...
```

That style seems easier to modify without getting errors for forgetting a backslash. Thoughts?

Reply:
Yeah, looks better that way. I was thinking in C syntax, will change